hazelcast/src/main/java/com/hazelcast/internal/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -1222,6 +1217,14 @@ public PartitionTableView createPartitionTableView() {
         private PartitionRuntimeState newState;
 
         public void run() {
+            if (!node.getClusterService().getClusterState().isJoinAllowed()) {
```

Comment:
What if join is allowed - in the `NO_MIGRATION` state? Is it ok to fetch the partition table and repair the partition table in `RepairPartitionTableTask`?

Reply:
Important point here is whether or not modification of the partition table is allowed in a state. But unfortunately, this is not obvious in current cluster state definitions. Before `NO_MIGRATION` was introduced, it was clearer. 

In `NO_MIGRATION` state, partition table can be modified, since we allow promotions to the primary and reassigning replicas when a partition is lost. 

Maybe I should add a third flag to `ClusterState` to denote if partition table can be modified or not.

Reply:
Just to confirm, will you be adding the third flag or is this PR complete then? You don't need to do it now, just to know if I can confirm this PR as it is.

Reply:
I started adding that but haven't finished yet. I'll try to push it today.

Reply:
Done.